api/src/utils/passport-setup.js

```diff
@@ -33,17 +30,32 @@ passport.use(
 
       if (user) {
         // user exists
-        return cb(null, user);
+        cb(null, user);
       } else {
-        const newUser = await new Member({
-          firstName: profile.name.givenName,
-          lastName: profile.name.familyName,
-          oauthID: profile.id,
+        // try to find a user with the same email
+        const unlinkedUser = await Member.findOne({
           email: profile.emails[0].value,
-          level: DEFAULT_LEVEL,
-        }).save();
+          oauthID: null,
```

Comment:
oauthID can't be null since it's unique. Instead of assigning the imported data an oauthID of null, we don't give them an oauthID at all (this is why it is marked as "sparse" in the schema). You can filter for objects that don't have an oauthID following this example: https://www.codegrepper.com/code-examples/c/mongoose+find+if+field+does+not+exists

Reply:
Thanks Matt, I looked into this and I think the `oauthID: null` query should reference both documents that contain a null `oauthID` and a missing one. (https://docs.mongodb.com/manual/tutorial/query-for-null-fields/)

Reply:
Let me know if you think it's better to query specifically for does not exist.

Reply:
Ok, this looks good.... no one should ever have an oauthID with a value of null anyway.